lib/matplotlib/cbook/__init__.py

```diff
@@ -584,18 +584,15 @@ class maxdict(dict):
     This doesn't override all the relevant methods to constrain the size,
     just ``__setitem__``, so use with caution.
     """
+
     def __init__(self, maxsize):
-        dict.__init__(self)
+        super().__init__()
         self.maxsize = maxsize
-        self._killkeys = []
 
     def __setitem__(self, k, v):
-        if k not in self:
-            if len(self) >= self.maxsize:
-                del self[self._killkeys[0]]
-                del self._killkeys[0]
-            self._killkeys.append(k)
-        dict.__setitem__(self, k, v)
+        super().__setitem__(k, v)
+        while len(self) >= self.maxsize:
+            del self[next(iter(self))]
```

Comment:
Does this ever delete more than one item? If so a `while` seems a bit overkill, but it's not a blocking issue.

Reply:
It's not clear whether e.g. `dict.update` calls `__setitem__` or not (see e.g. https://doc.pypy.org/en/latest/cpython_differences.html#subclasses-of-built-in-types).  The possibility of bypassing `__setitem__` is actually hinted at in the docstring.  For the use case of maxdict that's actually fine (that just means entries can get cached longer than expected), but that means that in theory you *could* have much more than `maxsize` items.
Given that it adds no complication to, in that case, fully shrink (on the next call to `__setitem__`) until hitting maxsize (it's just `while` vs `if`), we may just as well handle that...